posydon/binary_evol/MESA/step_mesa.py

```diff
@@ -788,31 +789,22 @@ def update_properties_NN(self, star_1_CO=False, star_2_CO=False,
 
         # update nearest neighbor core collapse quantites
         if interpolation_class != 'unstable_MT':
-            for prescrition in ['direct', 'Fryer+12-rapid', 'Fryer+12-delayed',
-                                'Sukhbold+16-engineN20',
-                                'Patton&Sukhbold20-engineN20']:
+            for MODEL_NAME in MODELS.keys():
                 for i, star in enumerate(stars):
                     if not stars_CO[i]:
-                        state = cb.final_values['S%d_%s_state'
-                                                % (i+1, prescrition)]
-                        SN_type = cb.final_values['S%d_%s_SN_type'
-                                                  % (i+1, prescrition)]
-                        f_fb = cb.final_values['S%d_%s_f_fb'
-                                               % (i+1, prescrition)]
-                        mass = cb.final_values['S%d_%s_mass'
-                                               % (i+1, prescrition)]
-                        spin = cb.final_values['S%d_%s_spin'
-                                               % (i+1, prescrition)]
-                        key = prescrition.replace('+', '')
-                        key = key.replace('-', '_')
-                        key = key.replace('&', '_')
+                        values = {}
+                        for key in ['state', 'SN_type', 'f_fb', 'mass', 'spin',
+                                    'm_disk_accreted', 'm_disk_radiated']:
+                            if key == 'state': 
+                                key = 'CO_type'
+                            values[key] = cb.final_values[f'S{i+1}_{MODEL_NAME}_{key}']
+                        state = cb.final_values[f'S{i+1}_{MODEL_NAME}_CO_type'] 
                         if state is None or state == 'None':
                             # privent to any quantities for star that did not
                             # reach core collpase
                             setattr(star, key, None)
```

Comment:
Here `key` is not well defined, because it is outside the for loop

Reply:
Good catch, I fixed the bug. This portion of the code was not tested as I did not run a population synthesis model with first nearest neighbour.